runtime/weak.c

```diff
@@ -537,10 +574,33 @@ CAMLexport void caml_ephemeron_blit_key(value ars, mlsize_t offset_s,
   offset_s += CAML_EPHE_FIRST_KEY;
   offset_d += CAML_EPHE_FIRST_KEY;
 
+  if ( caml_gc_phase == Phase_mark
+       && caml_ephe_list_pure
+       && Field(ard, CAML_EPHE_DATA_OFFSET) != caml_ephe_none
+       && !Is_white_val(ard)
+       ){
+    /* If the destination is in set (2) it is because of a white key */
```

Comment:
I don't understand this comment. Couldn't it be in set (2) because its data is black?